src/System.Private.ServiceModel/src/Internals/System/Runtime/IOThreadScheduler.cs

```diff
@@ -0,0 +1,664 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics.Contracts;
+using System.Runtime.InteropServices;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.Runtime
+{
+    internal class IOThreadScheduler
```

Comment:
Came from full framework class of the same name. Everything is identical except how we launch a thread to process the work items. Instead of posting a callback to another thread using IOOverlapped's I'm using TaskScheduler managed threads which aren't part of the ThreadPool. Basically the lines which say `Task.Factory.StartNew` used to call a different method to create a worker thread.